lib/breakfast/type.ex

```diff
@@ -384,21 +386,21 @@ defmodule Breakfast.Type do
         end
 
       {key_type, value_type}, acc ->
-        Enum.any?(map, fn {key, value} ->
+        Enum.reduce_while(map, [], fn {key, value}, acc ->
           case {validate(key_type, key), validate(value_type, value)} do
             {[], []} ->
-              true
+              {:halt, :ok}
 
-            _other ->
-              false
+            {key_errors, value_errors} ->
+              {:cont, key_errors ++ value_errors ++ acc}
           end
         end)
         |> case do
-          true ->
-            acc
-
-          false ->
+          :ok ->
             []
+
+          errors ->
+            errors ++ acc
```

Comment:
This was the bug I found :stuck_out_tongue: 

In the previous code, the `Enum.any?/2` should return `true` when a field was determined to be valid in the map, but the `false` case was returning a `[]` which meant that an invalid map would result in the errors being wiped

Reply:
Oh nice catch!